Kernel/Devices/HID/USB/KeyboardDevice.cpp

```diff
@@ -0,0 +1,1228 @@
+            if (m_num_lock_on) {
+                if (m_modifiers & Mod_Shift) {
+                    event.key = ShiftedKeyCodes[packet_raw.keypress5].key_code;
+                    event.map_entry_index = ShiftedKeyCodes[packet_raw.keypress5].map_entry_index;
+                    if (packet_raw.keypress5 == 0x54)
+                        event.code_point = '/';
+                } else {
+                    event.key = KeyCodes[packet_raw.keypress5].key_code;
+                    event.map_entry_index = KeyCodes[packet_raw.keypress5].map_entry_index;
+                    if (packet_raw.keypress5 == 0x54)
+                        event.code_point = '/';
+                }
+            } else {
+                if (m_modifiers & Mod_Shift) {
+                    event.key = ShiftedNumKeyCodes[packet_raw.keypress5].key_code;
+                    event.map_entry_index = ShiftedNumKeyCodes[packet_raw.keypress5].map_entry_index;
+                    if (packet_raw.keypress5 == 0x54)
+                        event.code_point = '/';
+                } else {
+                    event.key = NumKeyCodes[packet_raw.keypress5].key_code;
+                    event.map_entry_index = NumKeyCodes[packet_raw.keypress5].map_entry_index;
+                    if (packet_raw.keypress5 == 0x54)
+                        event.code_point = '/';
+                }
+            }
+            m_key5_pressed = true;
+            m_last_event = event;
+            handle_input_event(event);
+        }
+        if (packet_raw.keypress6 <= 0xe7 && !!packet_raw.keypress6 && !m_key6_pressed) {
+            event.flags |= Is_Press;
+            if (m_num_lock_on) {
+                if (m_modifiers & Mod_Shift) {
+                    event.key = ShiftedKeyCodes[packet_raw.keypress6].key_code;
+                    event.map_entry_index = ShiftedKeyCodes[packet_raw.keypress6].map_entry_index;
+                    if (packet_raw.keypress6 == 0x54)
+                        event.code_point = '/';
+                } else {
+                    event.key = KeyCodes[packet_raw.keypress6].key_code;
+                    event.map_entry_index = KeyCodes[packet_raw.keypress6].map_entry_index;
+                    if (packet_raw.keypress6 == 0x54)
+                        event.code_point = '/';
+                }
+            } else {
+                if (m_modifiers & Mod_Shift) {
+                    event.key = ShiftedNumKeyCodes[packet_raw.keypress6].key_code;
+                    event.map_entry_index = ShiftedNumKeyCodes[packet_raw.keypress6].map_entry_index;
+                    if (packet_raw.keypress6 == 0x54)
+                        event.code_point = '/';
+                } else {
+                    event.key = NumKeyCodes[packet_raw.keypress6].key_code;
+                    event.map_entry_index = NumKeyCodes[packet_raw.keypress6].map_entry_index;
+                    if (packet_raw.keypress6 == 0x54)
+                        event.code_point = '/';
```

Comment:
Just from a cursory glance: you repeat the below pattern a lot of times even though it doesn't matter which one of the four if-branches it takes:

```cpp
if (packet_raw.keypressX == 0x54)
    event.code_point = '/';
```

You should lift these up out of the branches and put it at the end.

Reply:
Fixed that